src/error.rs

```diff
@@ -16,6 +16,9 @@ pub enum Error {
     #[fail(display = "{:?}", _0)]
     Message(String),
 
+    #[fail(display = "{:?}", _0)]
+    MessageChain(String, #[cause] failure::Error, failure::Backtrace),
```

Comment:
I don't underestand "MessageChain" as an error type.  Is this to get more backtraces or something?

Reply:
This is like `Message`, it's a generic error type with only a string, but it adds the cause (another error instance i.e. a "chained" error) and backtrace.  And the cause/chained error is where I drew the naming from.  So yes this is using new name so I can add extra info.
As far as API design goes I consider this whole error enum not very good and hope that we'll eventually not have it anymore as functions return more specific things.  But to limit the refactoring impact when converting functions it's nice for now not have to change callers which use the existing `error::Error` type and being able to propagate things as a simple message but without throwing away the cause or backtrace.

Reply:
renamed